### PR TITLE
optimization changes

### PR DESCRIPTION
**Pull request by NitishMalik**
1. Added React Lazy
2. AsyncComponent
3. Added Header and Message Components

### CODE REVIEW

File: src/components/CardList.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import Card from "./Card";
 
 const CardList = ({ robos }) => {
+  console.log("Render Card List");
```
Comment (anjali2589) [inline src/components/CardList.jsx:5]: I think this was mistakenly left in the code.

File: src/components/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from "react";
+
+class Header extends Component {
+  shouldComponentUpdate() {
```
Comment (anjali2589) [inline src/components/Header.jsx:4]: Can we make a pure component instead of always sending false

File: src/components/CardList.jsx (hunk 1 above)
Comment (NitishMalik) [reply src/components/CardList.jsx:5]: Thanks, will correct it in next commit.

**Comment by NitishMalik** [comment]
Merging optimization changes to master

**Pull request merged**